catroid/src/androidTest/java/org/catrobat/catroid/test/content/bricks/CloneCompositeBrickUpdateVariableTest.java

```diff
@@ -0,0 +1,177 @@
+	public String name;
+
+	@Parameterized.Parameter(1)
+	public Brick brick;
+
+	private Sprite sprite;
+	private Script script;
+
+	@Before
+	public void setUp() {
+		Project project = new Project();
+		Scene scene = new Scene();
+		sprite = new Sprite("test_sprite");
+		scene.addSprite(sprite);
+		script = new StartScript();
+		sprite.addScript(script);
+		sprite.addUserVariable(USER_VARIABLE);
+		sprite.addUserList(USER_LIST);
+
+		project.addScene(scene);
+		ProjectManager.getInstance().setCurrentProject(project);
+
+		if (brick instanceof UserListBrick) {
+			((UserListBrick) brick).setUserList(USER_LIST);
+		}
+	}
+
+	@Test
+	public void testIf() {
+		IfLogicBeginBrick ifBrick = new IfLogicBeginBrick();
+		ifBrick.addBrickToIfBranch(brick);
+		script.addBrick(ifBrick);
+
+		Sprite cloneSprite = new SpriteController().copyForCloneBrick(sprite);
+		Brick cloneIfLogicBrick = cloneSprite.getScript(0).getBrick(0);
+		checkClonedBrickInNestedBrick((CompositeBrick) cloneIfLogicBrick);
+	}
+
+	@Test
+	public void testElse() {
+		IfLogicBeginBrick ifBrick = new IfLogicBeginBrick();
+		ifBrick.addBrickToElseBranch(brick);
+		script.addBrick(ifBrick);
+
+		Sprite cloneSprite = new SpriteController().copyForCloneBrick(sprite);
+		Brick cloneIfLogicBrick = cloneSprite.getScript(0).getBrick(0);
+		checkClonedBrickInNestedBrick((CompositeBrick) cloneIfLogicBrick);
+	}
+
+	@Test
+	public void testForever() {
+		ForeverBrick foreverBrick = new ForeverBrick();
+		foreverBrick.addBrick(brick);
+		script.addBrick(foreverBrick);
+
+		Sprite cloneSprite = new SpriteController().copyForCloneBrick(sprite);
+		Brick cloneForeverBrick = cloneSprite.getScript(0).getBrick(0);
+		checkClonedBrickInNestedBrick((CompositeBrick) cloneForeverBrick);
+	}
+
+	@Test
+	public void testRepeat() {
+		RepeatBrick repeatBrick = new RepeatBrick();
+		repeatBrick.addBrick(brick);
+		script.addBrick(repeatBrick);
+
+		Sprite cloneSprite = new SpriteController().copyForCloneBrick(sprite);
+		Brick cloneRepeatBrick = cloneSprite.getScript(0).getBrick(0);
+		checkClonedBrickInNestedBrick((CompositeBrick) cloneRepeatBrick);
+	}
+
+	@Test
+	public void doubleNested() {
+		RepeatBrick repeatBrick0 = new RepeatBrick();
+		RepeatBrick repeatBrick1 = new RepeatBrick();
+		repeatBrick1.addBrick(brick);
+		repeatBrick0.addBrick(repeatBrick1);
+
+		script.addBrick(repeatBrick0);
+		Sprite cloneSprite = new SpriteController().copyForCloneBrick(sprite);
+
+		Brick cloneRepeatBrick0 = cloneSprite.getScript(0).getBrick(0);
+		Brick cloneRepeatBrick1 = ((RepeatBrick) cloneRepeatBrick0).getNestedBricks().get(0);
+
+		checkClonedBrickInNestedBrick((CompositeBrick) cloneRepeatBrick1);
+	}
+
+	private void checkClonedBrickInNestedBrick(CompositeBrick compositeBrick) {
+		Brick nestedBrick = compositeBrick.getNestedBricks().size() == 0
+				? compositeBrick.getSecondaryNestedBricks().get(0)
+				: compositeBrick.getNestedBricks().get(0);
+
+		if (nestedBrick instanceof UserVariableBrick) {
```

Comment:
please put this as a new test method into TwoPartCompositeBrickTreeBuilderTest and ThreePart....